components/dashboard-modules/EnergyRecommendation.tsx

```diff
@@ -0,0 +1,75 @@
+import Image from 'next/image';
+import { FC } from 'react';
+import { useSiteData } from '~/lib/hooks';
+import { getCurrentTimeForecast } from '~/lib/utils';
+import content from '../../content/power-card-content.json';
+import NumberDisplay from './NumberDisplay';
+
+/**
+ * Determines the appliance with the greatest energy required that is less than or equal to the current output
+ * @param currentOutput the current output in kW based on the getCurrentTimeForecast
+ * @returns the index of the appliance with the greatest enrgy requiement less than or equal to the current ouput
+ * or -1 if an index could not be found
+ */
+
+const getBestRecommendationIndex = (currentOutput: number) => {
+  if (currentOutput != undefined) {
```

Comment:
No need to check for undefined here since `currentOutput` is of type `number` (which excludes `undefined` unless included explicitly). That's the power of TypeScript™.